services/skus/service.go

```diff
@@ -1583,3 +1584,209 @@ func (s *Service) UpdateOrderStatusPaidWithMetadata(ctx context.Context, orderID
 
 	return commit()
 }
+
+func (s *Service) CreateOrder(ctx context.Context, req *model.CreateOrderRequestNew) (*Order, error) {
+	items, err := createOrderItems(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check for number of items to be at least one.
+	//
+	// Validation should already have taken care of this.
+	// However, this method does not know about it.
+	// Therefore, an explicit check is necessary.
+	if len(items) == 0 {
+		return nil, model.ErrInvalidOrderRequest
+	}
+
+	const merchID = "brave.com"
+
+	numIntervals, err := s.createOrderIssuers(ctx, merchID, items)
+	if err != nil {
+		return nil, err
+	}
+
+	totalCost := model.OrderItemList(items).TotalCost()
+
+	status := model.OrderStatusPending
+	if totalCost.IsZero() {
+		status = model.OrderStatusPaid
+	}
+
+	// Use validFor from the first item.
+	//
+	// TODO: Deprecate the use of valid_for:
+	// valid_for_iso is now used instead of valid_for for calculating order's expiration time.
+	//
+	// The old code in CreateOrderFromRequest does a contradictory thing – it takes validFor from last item.
+	// It does not make any sense, but it's working because there is only one item normally.
+	var validFor time.Duration
+	if items[0].ValidFor != nil {
+		validFor = *items[0].ValidFor
+	}
+
+	order, err := s.Datastore.CreateOrder(
+		totalCost,
+		merchID,
+		status,
+		req.Currency,
+		// FIXME: Location.
+		//
+		// The old code in CreateOrderFromRequest contradictory things:
+		// - it looks as though it supports multiple items (mind the loop)
+		// - it requires all items to have the same location, at the same time.
+		// For this to work with bundles, this has to change.
+		// At this stage (i.e. just adding this new endpoint and switching over to it)
+		// using the location of the first (and the only) item accomplishes the same result.
+		items[0].Location.String,
+		&validFor,
+		items,
+		req.PaymentMethods,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create order: %w", err)
+	}
+
+	if !order.IsPaid() && order.IsStripePayable() {
+		if err := s.createStripeSessID(ctx, req, order); err != nil {
+			return nil, err
+		}
+	}
+
+	if numIntervals > 0 {
+		if err := s.Datastore.AppendOrderMetadataInt(ctx, &order.ID, "numIntervals", numIntervals); err != nil {
+			return nil, fmt.Errorf("failed to update order metadata: %w", err)
+		}
+	}
+
+	if err := s.Datastore.AppendOrderMetadataInt(ctx, &order.ID, "numPerInterval", 2); err != nil {
+		return nil, fmt.Errorf("failed to update order metadata: %w", err)
+	}
+
+	return order, nil
+}
+
+func (s *Service) createOrderIssuers(ctx context.Context, merchID string, items []model.OrderItem) (int, error) {
```

Comment:
As part of the refactoring new work, is it possible to bypass the create issuer checks for each order item by creating the issuer upfront, given that issuers are based on the encoded merchant and sku string e.g. [here](https://github.com/brave-intl/bat-go/blob/070c431cdadba86cd0114b1e92b992cc11637826/services/skus/credentials.go#L499).

Reply:
Added a todo-comment:

https://github.com/brave-intl/bat-go/blob/5be5d10d2380cc1a47bfec62cd43a3d9f7541b8b/services/skus/service.go#L1670-L1674